auto-value-firebase/src/main/java/me/mattlogan/auto/value/firebase/AutoValueFirebaseExtension.java

```diff
@@ -148,7 +182,11 @@ static MethodSpec generateStandardAutoValueConstructor(Map<String, TypeName> pro
       // This is important! This doesn't have to be here, but it's gotta be somewhere.
       checkIfTypeIsSupported(entry.getValue());
 
-      if (typeIsPrimitive(originalType) || typeIsPrimitiveCollection(originalType)) {
+      if(typeHasAdapter(originalType)){
+        ClassName output = getTypeAdapterOutputType(originalType);
+        fields.add(FieldSpec.builder(output, fieldName, PRIVATE).build());
+      }
+      else if (typeIsPrimitive(originalType) || typeIsPrimitiveCollection(originalType)) {
```

Comment:
Only other style nitpick (I think): can you put the `else`s on the same line as the `}`, like `} else if (thing) {`? There are a few of these.